lib/net/server/server.js

```diff
@@ -0,0 +1,35 @@
+'use strict'
+
+const EventEmitter = require('events')
+
+/**
+ * Base class for transport specific server implementations.
+ * @memberof module:net/server
+ */
+class Server extends EventEmitter {
+  get name () {
+    return this.constructor.name
+  }
+
+  /**
+   * Open server and get ready for new connections
+   * @protected
+   * @return {Promise} [description]
+   */
+  async open () {
+    throw new Error('Unimplemented')
+  }
+
+  /**
+   * Ban peer for a specified time
+   * @protected
+   * @param  {string} peerId id of peer
+   * @param  {number} [maxAge] how long to ban peer
+   * @return {Promise}
+   */
+  ban (peerId, maxAge) {
+    // don't do anything by default
+  }
+}
+
+module.exports = Server
```

Comment:
Had a (again rough) look over the different network classes. Have a first impression that these classes initiate good abstractions to serve as a basis in implement further (one :-) for the moment) protocols.

Don't have the complete picture on this yet if it will completely hold but this should serve as a solid basis to start with.

Should be taken care here where responsibilities lie between the classes here and the devp2p package and code is not replicated.